Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,342 @@
+    protected $connectionsData;
+
+    /**
+     * @var array
+     */
+    protected $amqpConnections = [];
+
+    /**
+     * Number of the port used by the broker
+     *
+     * @var int $port
+     */
+    protected $port;
+
+    /**
+     * Method responsible to catch the parameters to configure the driver.
+     *
+     * {@inheritDoc}
+     */
+    public function configure(string $host, string $username, string $password, string $vhost = null)
+    {
+        $this->connectionsData[] = [
+            'host' => $host,
+            'user' => $username,
+            'password' => $password,
+            'vhost' => $vhost,
+            'port' => $this->port
+        ];
+    }
+
+    /**
+     * Method responsible to open the connection with the broker.
+     *
+     * @param bool $shuffle - used to randomize the connections if exists more than one
+     *
+     * @throws \Exception
+     */
+    public function connect($shuffle = true)
+    {
+        if ($shuffle) {
+            $this->shuffleConnections();
+        }
+
+        try {
+            $this->addConnection(AMQPStreamConnection::create_connection($this->connectionsData, [
+                'insist' => true,
+                'login_method' => 'AMQPLAIN',
+                'locale' => 'en_UK',
+                'connection_timeout' => 60.0,
+                'read_write_timeout' => 50.0,
+                'context' => null,
+                'keepalive' => true,
+                'heartbeat' => 0,
+            ]));
+        } catch (AMQPIOException $exception) {
+            throw new \AMQPConnectionException($exception->getMessage());
+        }
+    }
+
+    /**
+     * Add a connection to the array of connections.
+     *
+     * @param $connection
+     */
+    public function addConnection(AbstractConnection $connection)
+    {
+        $this->amqpConnections[] = $connection;
+    }
+
+    /**
+     * Responsible to disconnect with the broker.
+     *
+     * @throws \Exception
+     */
+    public function disconnect()
+    {
+        foreach ($this->amqpConnections as $connection) {
+            if ($connection instanceof AbstractConnection) {
+                $connection->close();
+            }
+        }
+    }
+
+    /**
+     * Creates the QueueMessage object.
+     *
+     * @return QueueMessageInterface
+     */
+    public function createQueueMessage(): QueueMessageInterface
+    {
+        $msg = new QueueMessage();
+        $msg->setContext(new Context());
+
+        return $msg;
+    }
+
+    /**
+     * Verifies if there is some connection opened with the broker.
+     */
+    public function isConnected(): bool
+    {
+        foreach ($this->amqpConnections as $connection) {
+            if ($connection->is_connected()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function destroy()
+    {
+        $this->amqpConnections = [];
+        $this->connectionsData = [];
```

Comment:
Is it needed to setup both properties to `[]`? And when this method is trigggered?

Reply:
In the application in the test methods. But, in this case the destroy needs to do it to re-initialize the variable, otherwise the connection will be close but still available in the amqpConnection variable that turns possible to make the connection again.